### PR TITLE
refactor: update Next.js, React.js and Node.js

### LINKED ISSUE #794: jemalloc page size issue for arm64
### Description

building QPKG for Qnap NAS, while all work perfectily on x86_64, ARM64 has issue du jemmaloc page file size issue

image

### Version

1.9.0

### Steps to Reproduce

NODEJS 20
git version

git clone https://github.com/Fallenbagel/jellyseerr.git && cd jellyseerr
git checkout main

cd jellyseerr

CYPRESS_INSTALL_BINARY=0 CC=${HOST}-cc CXX=${HOST}-g++ npm_config_build_from_source=true npm_config_build_from_source=sqlite3 ../bin/yarn install --frozen-lockfile --network-timeout 1000000  ;
CC=${HOST}-cc CXX=${HOST}-g++ npm_config_build_from_source=true npm_config_build_from_source=sqlite3 ../bin/yarn run build ;

### Screenshots

_No response_

### Logs

_No response_

### Platform

desktop

### Device

QNAP NAS 

### Operating System

Linux ARM64

### Browser

Firefox

### Additional Context

_No response_

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

### PR DESCRIPTION
**Pull request by gauthier-th**
#### Description

This PR updates Next.js, React.js and Node.js to their latest version.

#### To-Dos

- [x] Successful build `yarn build`
- [x] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #794

### CODE REVIEW

**Comment by gauthier-th** [comment]
> found a small bug with the requester profile icon not being resized correctly under recent requests.
> 
> 342007805-d77242fb-f5af-40f0-a473-27469a0710fb

This should be fixed

**Comment by Gauvino** [comment]
First bug on request page the collection at the too same button with one not being displayed properly.
chrome_fifRtACbbJ
Second one appears not on everyone, in the request menu some picture of profile I think of requester are not rendered properly and break the request card. (Note: the render of broken profile picture only occurred in request car on request slide)
chrome_GZozC4Ik0P
chrome_oZVlqrTIQc

And that's is for the graphical side I think, I haven't encountered any other problems !

Thank you for all this amazing work really !

**Comment by gauthier-th** [comment]
> First bug on request page the collection at the too same button with one not being displayed properly. chrome_fifRtACbbJ Second one appears not on everyone, in the request menu some picture of profile I think of requester are not rendered properly and break the request card. (Note: the render of broken profile picture only occurred in request car on request slide) chrome_GZozC4Ik0P chrome_oZVlqrTIQc
> 
> And that's is for the graphical side I think, I haven't encountered any other problems !
> 
> Thank you for all this amazing work really !

Thanks for reporting this! It should be fixed now.

**Comment by Gauvino** [comment]
Bug fixed, and everything seems to be working fine, interaction, menu, card, everything is great.
Thanks for everything

File: Dockerfile.local [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:18.18-alpine
+FROM node:20-alpine
```
Comment (fallenbagel) [inline Dockerfile.local:1]: Migrate to pnpm

File: README.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ https://hub.docker.com/r/fallenbagel/jellyseerr
 
 Pre-requisites:
 
-- Nodejs [v18](https://nodejs.org/download/release/v18.18.2)
+- Nodejs [v20](https://nodejs.org/en/download)
 - [Yarn](https://classic.yarnpkg.com/lang/en/docs/install)
```
Comment (fallenbagel) [inline README.md]: Link to pnpm

File: README.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ https://hub.docker.com/r/fallenbagel/jellyseerr
...
 - Download/git clone the source code from the github (Either develop branch or main for stable)
```
Comment (fallenbagel) [inline README.md:57]: Migrate to pnpm

File: README.md [hunk 4]
```diff
@@ -73,7 +73,7 @@ _To set env variables such as `JELLYFIN_TYPE=emby` create a file called `.env` i
 
 **Pre-requisites:**
 
-- Nodejs [v18](https://nodejs.org/en/download/package-manager)
+- Nodejs [v20](https://nodejs.org/en/download)
 - [Yarn](https://classic.yarnpkg.com/lang/en/docs/install) (on Debian based distros, the package manager provided `yarn` is different and is a package called cmdlet. You can remove that using `apt-remove cmdlet` then install yarn using `npm install -g yarn`)
```
Comment (fallenbagel) [inline README.md]: Link to pnpm

File: src/components/IssueBlock/index.tsx [hunk 5]
```diff
@@ -64,7 +63,7 @@ const IssueBlock = ({ issue }: IssueBlockProps) => {
           </div>
         </div>
         <div className="ml-2 flex flex-shrink-0 flex-wrap">
-          <Link href={`/issues/${issue.id}`} passHref>
+          <Link href={`/issues/${issue.id}`} passHref legacyBehavior>
```
Comment (fallenbagel) [inline src/components/IssueBlock/index.tsx:66]: @gauthier-th why was legacyBehavior prop added to this when `<a>` has been removed?

> An <a> element is no longer required as a child of <Link>. Add the legacyBehavior prop to use the legacy behavior or remove the <a> to upgrade.

File: src/components/IssueList/IssueItem/index.tsx [hunk 6]
```diff
@@ -259,7 +262,7 @@ const IssueItem = ({ issue }: IssueItemProps) => {
       </div>
       <div className="z-10 mt-4 flex w-full flex-col justify-center pl-4 pr-4 xl:mt-0 xl:w-96 xl:items-end xl:pl-0">
         <span className="w-full">
-          <Link href={`/issues/${issue.id}`} passHref>
+          <Link href={`/issues/${issue.id}`} passHref legacyBehavior>
```
Comment (fallenbagel) [inline src/components/IssueList/IssueItem/index.tsx:265]: See previous comment

File: src/components/IssueModal/CreateIssueModal/index.tsx [hunk 7]
```diff
@@ -118,7 +119,7 @@ const CreateIssueModal = ({
                     strong: (msg: React.ReactNode) => <strong>{msg}</strong>,
                   })}
                 </div>
-                <Link href={`/issues/${newIssue.data.id}`}>
+                <Link href={`/issues/${newIssue.data.id}`} legacyBehavior>
```
Comment (fallenbagel) [inline src/components/IssueModal/CreateIssueModal/index.tsx:122]: Same comment as above

File: src/components/Login/LocalLogin.tsx [hunk 8]
```diff
@@ -137,7 +138,7 @@ const LocalLogin = ({ revalidate }: LocalLoginProps) => {
                   </span>
                   {passwordResetEnabled && (
                     <span className="inline-flex rounded-md shadow-sm">
-                      <Link href="/resetpassword" passHref>
+                      <Link href="/resetpassword" passHref legacyBehavior>
```
Comment (fallenbagel) [inline src/components/Login/LocalLogin.tsx:141]: See above

File: src/components/ResetPassword/RequestResetLink.tsx [hunk 9]
```diff
@@ -66,7 +70,7 @@ const ResetPassword = () => {
                   {intl.formatMessage(messages.requestresetlinksuccessmessage)}
                 </p>
                 <span className="mt-4 flex justify-center rounded-md shadow-sm">
-                  <Link href="/login" passHref>
+                  <Link href="/login" passHref legacyBehavior>
```
Comment (fallenbagel) [inline src/components/ResetPassword/RequestResetLink.tsx:73]: See above

File: src/components/ResetPassword/index.tsx [hunk 10]
```diff
@@ -81,7 +85,7 @@ const ResetPassword = () => {
                   {intl.formatMessage(messages.resetpasswordsuccessmessage)}
                 </p>
                 <span className="mt-4 flex justify-center rounded-md shadow-sm">
-                  <Link href="/login" passHref>
+                  <Link href="/login" passHref legacyBehavior>
```
Comment (fallenbagel) [inline src/components/ResetPassword/index.tsx:88]: Above

File: src/components/Settings/Notifications/NotificationsWebhook/index.tsx [hunk 11]
```diff
@@ -309,6 +313,7 @@ const NotificationsWebhook = () => {
                   <Link
                     href="https://docs.overseerr.dev/using-overseerr/notifications/webhooks#template-variables"
                     passHref
+                    legacyBehavior
```
Comment (fallenbagel) [inline src/components/Settings/Notifications/NotificationsWebhook/index.tsx:316]: Above

File: src/components/UserProfile/ProfileHeader/index.tsx [hunk 12]
```diff
@@ -88,6 +91,7 @@ const ProfileHeader = ({ user, isSettingsPage }: ProfileHeaderProps) => {
                 : `/users/${user.id}/settings`
             }
             passHref
+            legacyBehavior
```
Comment (fallenbagel) [inline src/components/UserProfile/ProfileHeader/index.tsx:94]: Same as above

File: src/components/UserProfile/ProfileHeader/index.tsx [hunk 13]
```diff
@@ -101,6 +105,7 @@ const ProfileHeader = ({ user, isSettingsPage }: ProfileHeaderProps) => {
                 loggedInUser?.id === user.id ? `/profile` : `/users/${user.id}`
               }
               passHref
+              legacyBehavior
```
Comment (fallenbagel) [inline src/components/UserProfile/ProfileHeader/index.tsx:108]: Above

File: src/components/IssueBlock/index.tsx (hunk 5 above)
Comment (gauthier-th) [reply src/components/IssueBlock/index.tsx:66]: See the first child, Button? It is a `<a>`, so we need to keep the legacy behavior

**Review by Fallenbagel — DISMISSED** [review]

**Review by Fallenbagel — APPROVED** [review]
LGTM!

**Pull request merged**